code/game/objects/structures/crates_lockers/largecrate.dm

```diff
@@ -18,16 +18,36 @@
 	return
 
 /obj/structure/largecrate/proc/unpack()
-	for(var/atom/movable/A in contents)
-		A.forceMove(loc)
+	var/turf/current_turf = get_turf(src) // Get the turf the crate is on
+
 	playsound(src, unpacking_sound, 35)
+
+	/// Store the reference of the crate material
+	var/obj/item/stack/sheet/material_sheet
+	if(parts_type) // Create the crate material and store its reference
+		material_sheet = new parts_type(current_turf, 2)
+
+	// Move the objects back to the turf, above the crate material
+	while(contents.len)
```

Comment:
why this over for(var/atom/movable/moving_atom in contents)?

Reply:
> why this over for(var/atom/movable/moving_atom in contents)?

Just what I decided to go with while I was typing it up.
In my mind there might not be contents to process or there might be a lot of contents so I just put that in a while.

Reply:
> > why this over for(var/atom/movable/moving_atom in contents)?
> 
> Just what I decided to go with while I was typing it up. In my mind there might not be contents to process or there might be a lot of contents so I just put that in a while.
> 
> Do I need to change it?

iterating over a list with a for loop is slightly better optimized, and is preferred in our style guide. while loops can be finicky in dm and cause the whole mc to grind to a stop. if there's a lot of contents, or no contents, a for loop will handle it just as well

Reply:
> for(var/atom/movable/moving_atom in contents)

No worries; updated.